### ISSUE TITLE
Bug report: Copy button breaks (persists copied state) on double click

### ISSUE THREAD

**Issue by kristinchaya** [hashicorp/design-system#1052]
## IS: Current behavior
When double clicking the [copy button](https://github.com/hashicorp/design-system/blob/main/website/app/components/doc/copy-button/index.js), the copied state persists and the timeout breaks.

https://user-images.githubusercontent.com/120415408/212147944-915366c0-e48a-4c54-aee2-d743d2d3fe1f.mov

## SHOULD BE: Expected behavior
Double click behavior should be disregarded or handled as a single click.

## Steps to reproduce
1. Double click on a "click to copy" button
2. Observe the copied state disappear and reappear, and then persist

**Comment by alex-ju** [comment]
Hi @kristinchaya! 👋  I can replicate the issue, but `cloud-ui` doesn't use design system components for the copy code functionality we see in this recording. If you meant to raise this issue in https://github.com/hashicorp/cloud-ui let us know and we can transfer it.

**Comment by kristinchaya** [comment]
@alex-ju yes, transferring it would be awesome! Sorry about that - I'm new 😄
